scheduler/scheduler-server/src/main/java/org/ow2/proactive/scheduler/synchronization/AOSynchronization.java

```diff
@@ -0,0 +1,824 @@
+            throws InvalidChannelException, CompilationException, IOException {
+        try {
+            Channel<T> chosenChannel = getChannel(channel);
+            chosenChannel.replaceAll((BiFunction<? super String, ? super T, ? extends T>) evaluateClosure(function,
+                                                                                                          BiFunction.class));
+            commitIfNeeded(channel);
+            logWithContext(taskid, channel, "Run replaceAll, new content is: " + chosenChannel, Level.DEBUG);
+        } catch (InvalidChannelException | CompilationException | IOException e) {
+            throw e;
+        } catch (Throwable e) {
+            throw new ClosureEvaluationException("Exception when evaluating closure : " +
+                                                 StackTraceUtil.getStackTrace(e));
+        }
+    }
+
+    @Override
+    public <T extends Serializable> T putIfAbsent(String originator, TaskId taskid, String channel, String key, T value)
+            throws InvalidChannelException, IOException {
+        T answer = ((Channel<T>) getChannel(channel)).putIfAbsent(key, value);
+        commitIfNeeded(channel);
+        logWithContext(taskid,
+                       channel,
+                       "Put " + value + " on key '" + key + "', previous value was " + answer,
+                       Level.DEBUG);
+        return answer;
+    }
+
+    @Override
+    public <T extends Serializable> boolean remove(String originator, TaskId taskid, String channel, String key,
+            T value) throws InvalidChannelException, IOException {
+        boolean answer = getChannel(channel).remove(key, value);
+        commitIfNeeded(channel);
+        logWithContext(taskid,
+                       channel,
+                       "Tried to remove key '" + key + "' with expected value " + value + ", success=" + answer,
+                       Level.DEBUG);
+        return answer;
+    }
+
+    @Override
+    public <T extends Serializable> boolean replace(String originator, TaskId taskid, String channel, String key,
+            T oldValue, T newValue) throws InvalidChannelException, IOException {
+        boolean answer = getChannel(channel).replace(key, oldValue, newValue);
+        commitIfNeeded(channel);
+        logWithContext(taskid,
+                       channel,
+                       "Tried to replace key '" + key + "' with expected old value " + oldValue + " and new value " +
+                                newValue + ", success=" + answer,
+                       Level.DEBUG);
+        return answer;
+
+    }
+
+    @Override
+    public <T extends Serializable> T replace(String originator, TaskId taskid, String channel, String key, T value)
+            throws InvalidChannelException, IOException {
+        T answer = ((Channel<T>) getChannel(channel)).replace(key, value);
```

Comment:
![MINOR](https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR') Remove this unnecessary cast to "Channel". [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS1905)